crates/re_viewport/src/viewport_blueprint_ui.rs

```diff
@@ -149,53 +147,53 @@ impl ViewportBlueprint<'_> {
         let mut visibility_changed = false;
         let mut action = TreeAction::Keep;
         let mut visible = self.tree.is_visible(tile_id);
+        let visible_child = visible;
         let item = Item::SpaceView(space_view.id);
-        let is_selected = ctx.selection().contains(&item);
 
         let root_group = space_view.contents.root_group();
         let default_open = Self::default_open_for_group(root_group);
         let collapsing_header_id = ui.id().with(space_view.id);
-        egui::collapsing_header::CollapsingState::load_with_default_open(
-            ui.ctx(),
-            collapsing_header_id,
-            default_open,
-        )
-        .show_header(ui, |ui| {
-            blueprint_row_with_buttons(
-                ctx.re_ui,
-                ui,
-                true,
-                visible,
-                is_selected,
-                |ui| {
-                    let response = crate::item_ui::space_view_button(ctx, ui, space_view);
-                    if response.clicked() {
-                        focus_tab(&mut self.tree, space_view_id);
-                    }
-                    response
-                },
-                |re_ui, ui| {
-                    visibility_changed =
-                        visibility_button_ui(re_ui, ui, true, &mut visible).changed();
-                    if re_ui
-                        .small_icon_button(ui, &re_ui::icons::REMOVE)
-                        .on_hover_text("Remove Space View from the Viewport")
-                        .clicked()
-                    {
-                        action = TreeAction::Remove;
-                    }
-                },
-            );
-        })
-        .body(|ui| {
-            Self::space_view_blueprint_ui(
-                ctx,
-                ui,
-                space_view.contents.root_handle(),
-                space_view,
-                visible,
-            );
-        });
+        let is_item_hovered =
+            ctx.selection_state().highlight_for_ui_element(&item) == HoverHighlight::Hovered;
+
+        let response = ListItem::new(ctx.re_ui, space_view.display_name.clone())
+            .selected(ctx.selection().contains(&item))
+            .subdued(!visible)
+            .override_hover(is_item_hovered)
```

Comment:
I wonder if we can come up with a more elegant solution to this. It's not obvious how though.

The best I can come up with is this: produce the `Id` of the list item first, then call `egui::Context::highlight_widget` (if needed) before calling `ListItem::new(…).id(id)`.

(…though this also requires a fix to egui to make `highlight_widget` take effect the same frame).

Alternative fix: add something to egui along the lines of `ui.highlight_next_item()`, but I don't really like that sort of statefullness (and it might accidentally affect only part of the next widget, e.g. the triangle).

Reply:
I don't like this much either, but I'm in favour of punting on that.

My rationale is that piggy-backing on the hover state to show relations between disjoints UI elements related to the same "thing" is just wrong in the first place. With the new `ListItem`, it's visually too intrusive for my taste (I have that noted in #3120):
<img width="1712" alt="image" src="https://github.com/rerun-io/rerun/assets/49431240/0192d812-796a-4c7e-9825-ffc5cd9acfd3">



In the long run, we will instead need a new kind of dedicated highlighting, which will require adapting the Selection Panel UI as well. This new state will be unknown to egui, so it's probably not the right place to hack now.

Reply:
Makes sense - though we could also make `highlight` something separate in egui from `hovered`. But I agree on the punt part